test-e2e.sh

```diff
@@ -81,6 +81,22 @@ kubectl version
 # Don't build/test the prow image as it requires Docker
 EXCLUDED_TARGETS="-//images/gcloud-bazel:gcloud_install -//images/gcloud-bazel:gcloud_push"
 
+# Create a unique namespace for this job using the repo name and the build id
+export E2E_NAMESPACE="${REPO_OWNER:-default-owner}-${REPO_NAME:-default-name}-${BUILD_ID:-0}"
+
+(cat <<EOF
+{
+  "kind": "Namespace",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "${E2E_NAMESPACE}",
+    "labels": {
+    }
+  }
+}
+EOF
+) |  kubectl create -f - || true
```

Comment:
This is probably easier as: `kubectl get "namespaces/${E2E_NAMESPACE}"  || kubectl create namespace "${E2E_NAMESPACE}"`

Whatever you do, it will be easier to debug namespace creation failures if you remove the `|| true` (or somehow validate that the namespace exists before continuing further.